tests/test_encoding/test_decision_tree_encoder.py

```diff
@@ -2,24 +2,96 @@
 import pandas as pd
 import pytest
 
+from sklearn.exceptions import NotFittedError
+
 from feature_engine.encoding import DecisionTreeEncoder
 
 
-def test_encoding_method_param(df_enc):
-    # defaults
+# init parameters
+@pytest.mark.parametrize("enc_method", ["count", False, 1])
+def test_error_if_encoding_method_not_permitted_value(enc_method):
+    msg = (
+        "`encoding_method` takes only values 'ordered' and 'arbitrary'."
+        f" Got {enc_method} instead."
+    )
+    with pytest.raises(ValueError, match=msg):
+        DecisionTreeEncoder(encoding_method=enc_method)
+
+
+@pytest.mark.parametrize(
+    "unseen", ["string", False, ("raise", "ignore"), ["ignore"], np.nan]
+)
+def test_error_if_unseen_gets_not_permitted_value(unseen):
+    msg = (
+        "Parameter `unseen` takes only values ignore, raise, encode. "
+        f"Got {unseen} instead."
+    )
+    with pytest.raises(ValueError) as record:
```

Comment:
Miss one

Reply:
This one should also be fixed

Reply:
Is this comment new? I don't think I left any record in the file I committed last. Just doubled checked and didnt find any left.

The error comes from the values within parametrize, when I pass a list or a tuple, say ["raise", "ignore"] and then that tuple is passed as {unseen}, it doesn't like it. But it resolved with re.escape(message here)